`msys2` is no longer needed on Windows?

It seems we do not need `msys2` dependency on Windows.
My local Windows machines have no `msys2` or `rtools` on their `PATH`s, I only use `RTOOLS40_HOME` environment variable, yet I am able to run all tests.

When performing dynamic compilation, we inject path to `rtools` here
https://github.com/extendr/rextendr/blob/09c830538e0f811ec49b4abaf4507870b476aa53/R/source.R#L225-L244
And when Rust code is part of the package and is compiled with `Makevars`, `rtools` is added to the `PATH` by R itself.
In both cases, we get the correct dependencies.

I removed these dependencies from CI [here](https://github.com/Ilia-Kosenkov/rextendr/tree/msys2-free-ci)
https://github.com/extendr/rextendr/blob/09c830538e0f811ec49b4abaf4507870b476aa53/.github/workflows/R-CMD-check.yaml#L72-L73
https://github.com/extendr/rextendr/blob/09c830538e0f811ec49b4abaf4507870b476aa53/.github/workflows/test_pkg_gen.yaml#L70-L71
and these are outputs from two primary CIs:
- [R CMD check](https://github.com/Ilia-Kosenkov/rextendr/actions/runs/668277668)
- [Integration tests](https://github.com/Ilia-Kosenkov/rextendr/actions/runs/668277662)

@Ilia-Kosenkov 
While the test on package generation works, the other test of R CMD check fails on my local without having msys2 on `PATH`. For the success of CI, my guess is that GitHub Actions CI includes msys2 by default whether we explicitly add it or not. But I have no idea about the difference between my setup and yours...

```
==> devtools::test()

Loading rextendr
Testing rextendr
�� |  OK F W S | Context
/ |   0       | eval                                                                                                         Updating crates.io index
   Compiling winapi-build v0.1.1
   Compiling winapi v0.3.9
   Compiling winapi v0.2.8
   Compiling proc-macro2 v1.0.24
   Compiling unicode-xid v0.2.1
   Compiling syn v1.0.65
   Compiling extendr-engine v0.2.0
   Compiling lazy_static v1.4.0
   Compiling kernel32-sys v0.2.2
   Compiling quote v1.0.9
   Compiling extendr-macros v0.2.0
   Compiling libR-sys v0.2.1
   Compiling extendr-api v0.2.0
   Compiling rextendr1 v0.0.1 (C:\Users\HIROAK~1\AppData\Local\Temp\RtmpKkE6Fn\file2248170e408d)
error: linking with `x86_64-w64-mingw32-gcc` failed: exit code: 1
  |
  = note: "x86_64-w64-mingw32-gcc" "-fno-use-linker-plugin" "-Wl,--nxcompat" "-Wl,--dynamicbase" "-Wl,--disable-auto-image-base" "-m64" "-Wl,--high-entropy-va" "C:\\Users\\hiroaki-yutani\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsbegin.o" "-L" ...snip...  "-Wl,-Bdynamic" "-lR" "-ladvapi32" "-lws2_32" "-luserenv" "-lgcc_eh" "-l:libpthread.a" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-luser32" "-lkernel32" "C:\\Users\\hiroaki-yutani\\.rustup\\toolchains\\stable-x86_64-pc-windows-msvc\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: C:/rtools40/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/8.3.0/../../../../x86_64-w64-mingw32/bin/ld.exe: cannot find -lgcc_eh
          collect2.exe: error: ld returned 1 exit status
          

error: aborting due to previous error

error: could not compile `rextendr1`
```

--- a/.github/workflows/R-CMD-check.yaml
+++ b/.github/workflows/R-CMD-check.yaml
@@ -38,10 +38,8 @@ jobs:
           toolchain: ${{ matrix.config.rust-version }}
           default: true
 
-      # Uses @master branch to address rtools path issue
-      # https://github.com/r-lib/actions/issues/228
       - name: Set up R
-        uses: r-lib/actions/setup-r@master
+        uses: r-lib/actions/setup-r@v1
         with:
           r-version: ${{ matrix.config.r }}
           windows-path-include-mingw: false

--- a/.github/workflows/R-CMD-check.yaml
+++ b/.github/workflows/R-CMD-check.yaml
@@ -67,8 +67,6 @@ jobs:
         run: |
           rustup target add x86_64-pc-windows-gnu
           rustup target add i686-pc-windows-gnu
-          echo "C:\msys64\mingw64\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append 
-          echo "C:\msys64\mingw32\bin" | Out-File -FilePath $env:GITHUB_PATH -Encoding utf8 -Append
         shell: pwsh
 
       - name: Configure Linux

